tests/core/nlg/test_response.py

```diff
@@ -80,6 +80,31 @@ async def test_nlg_when_multiple_conditions_satisfied():
     assert resp.get("text") in ["example A", "example B"]
 
 
+@pytest.mark.parametrize(("response_text"), ("example a",))
+async def test_nlg_slot_case_sensitivity(response_text: Text):
+    responses = {
+        "utter_action": [
+            {
+                "text": response_text,
+                "condition": [{"type": "slot", "name": "test", "value": "Cold"}],
```

Comment:
I would also leave this in lowercase to match the slot mapping potential values, then pass to the Slot initialisation of `initial_value` the string with the capital letter (`Cold`).

Reply:
Ya. Done with the required changes

Reply:
@ancalita The code quality tests fails after restructuring the nested if conditions. Do we ignore this test as of now ?